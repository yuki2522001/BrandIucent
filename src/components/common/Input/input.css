.input__value {
  background: var(--light);
  border-radius: 49.8249px;
  padding: 25px 0 25px 60px;
  width: 682px;
}