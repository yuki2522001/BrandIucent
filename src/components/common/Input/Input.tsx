import React, { memo } from "react";
import "./input.css";

interface placeholder {
  type: string;
  placeholder: string;
}

const Input: React.FC<placeholder> = memo(({ type, placeholder }) => {
  return (
    <input
      data-testid="input-value"
      className="input__value"
      type={type}
      placeholder={placeholder}
    />
  );
});

export default Input;
