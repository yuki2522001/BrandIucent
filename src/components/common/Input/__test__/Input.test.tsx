import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Input from "../Input";

describe("Input component", () => {
  test("should render Input component", () => {
    const { getByTestId } = render(<Input type="text" placeholder="Email address" />);
    expect(getByTestId("input-value")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<Input type="text" placeholder="Email address" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
