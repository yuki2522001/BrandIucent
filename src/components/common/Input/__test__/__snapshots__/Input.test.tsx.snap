// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component matches snapshot 1`] = `
<DocumentFragment>
  <input
    class="input__value"
    data-testid="input-value"
    placeholder="Email address"
    type="text"
  />
</DocumentFragment>
`;
